File: 0238-product-of-array-except-self/0238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] prefix = new int[nums.length];
        int[] suffix = new int[nums.length];
        Arrays.fill(prefix,1);
        Arrays.fill(suffix,1);
        int n = nums.length;

        for(int i = 1;i < n;i++){
            prefix[i] = prefix[i-1]*nums[i-1];
        }
        for(int i = n-2;i>=0;i--){
            suffix[i] = suffix[i+1]*nums[i+1];
        }

        for(int i = 0;i < n ;i++){
            nums[i] = prefix[i]*suffix[i];
        }
        return nums;
    }
}

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function(nums) {
    const prefix = new Array(nums.length).fill(1);
    const suffix = new Array(nums.length).fill(1);

    for(let i = 1;i < nums.length;i++){
        prefix[i] = nums[i-1]*prefix[i-1]; 
    }
    for(let i = nums.length-2;i >= 0;i--){
        suffix[i] = nums[i+1]*suffix[i+1]; 
    }
    for(let i = 0;i < nums.length;i++){
        nums[i] = prefix[i]*suffix[i];
    }
    return nums;
};